### DISCUSSION TITLE
Will there be, or is there a mac version for COSMOS?

### DISCUSSION THREAD

**Discussion by josiauh** [CosmosOS/Cosmos#2228] [Q&A]
While I **had** used COSMOS, the device i used won't work anymore, I have a mac alongside, which i am using now.
I want to use COSMOS again, but have no windows devices left.
Will there be a build for macOS, or does it already exist?

**Comment by AsertCreator** [comment]
if you're using apple silicon mac: you can use parralels but vs will not run. if you're using intel mac: use boot camp to install windows alongside macos then install vs and cosmos

**Reply by AsertCreator** [reply]
p.s. there is no mac build

**Reply by josiauh** [reply]
Thanks for letting me know!

**Comment by zarlo** [comment]
you can work on Cosmos on MacOS you need to use the crossplatform branch (note i dont know if its been tested on MacOS)

**Reply by josiauh** [reply]
I can't (and really don't want to) work on a mac version, so I don't think this will work.

**Reply by abjdev** [reply]
you need to install dotnet to mac

**Reply by josiauh** [reply]
> you need to install dotnet to mac

Do you mean like building it or running it?

**Reply by abjdev** [reply]
no, you need dotnet to install cosmos, try that [Dotnet for mac](https://dotnet.microsoft.com/en-us/download/dotnet/thank-you/sdk-5.0.408-macos-x64-installer)

**Reply by josiauh** [reply]
> no, you need dotnet to install cosmos, try that [Dotnet for mac](https://dotnet.microsoft.com/en-us/download/dotnet/thank-you/sdk-5.0.408-macos-x64-installer)

Isn't building made with a BAT file, which mac doesn't support?

**Reply by zarlo** [reply]
> > no, you need dotnet to install cosmos, try that [Dotnet for mac](https://dotnet.microsoft.com/en-us/download/dotnet/thank-you/sdk-5.0.408-macos-x64-installer)
> 
> Isn't building made with a BAT file, which mac doesn't support?

the crossplatform branch has a build system that will work on mac and linux

**Reply by josiauh** [reply]
> > > no, you need dotnet to install cosmos, try that [Dotnet for mac](https://dotnet.microsoft.com/en-us/download/dotnet/thank-you/sdk-5.0.408-macos-x64-installer)
> > 
> > 
> > Isn't building made with a BAT file, which mac doesn't support?
> 
> the crossplatform branch has a build system that will work on mac and linux

Yeah, I don't see that there. Is the build system in a folder, or do I just build from the SLN files?

**Reply by MishaTy** [reply]
It's a makefile

**Reply by josiauh** [reply]
> It's a makefile

/Library/Developer/CommandLineTools/usr/bin/make build
Building IL2CPU
dotnet build /Users/josiah/Downloads/Cosmos-crossplatform/../IL2CPU -v:q -nologo
make[1]: dotnet: No such file or directory
make[1]: *** [build] Error 1
make: *** [all] Error 2

**Reply by zarlo** [reply]
you need to install .net 5

**Reply by josiauh** [reply]
> you need to install .net 5
Sorry for the late reply, I installed net 6, checked by "dotnet --info"

**Reply by josiauh** [reply]
Error would be fixed.
(I found out why.)

**Reply by josiauh** [reply]
@MishaTY Make completed. What's next?
